### ISSUE TITLE
Destructuring two variables and reassigning the value of one is hard with let and const

### ISSUE THREAD

**Issue by ericandrewlewis** [eslint/eslint#8187]
**Tell us about your environment**

* **ESLint Version:** 3.16.1
* **Node Version:** v6.6.0
* **npm Version:** 3.10.3

**What parser (default, Babel-ESLint, etc.) are you using?** default

**Please show your full configuration:**

```js
module.exports = {
  "env": {
    "browser": true,
    "es6": true
  },
  "extends": "eslint:recommended",
  "parserOptions": {
    "sourceType": "module",
    "ecmaFeatures": {
      "experimentalObjectRestSpread": true
    }
  },
  "rules": {
    "prefer-const": ["error", {
      "destructuring": "any",
      "ignoreReadBeforeAssign": false
    }],
    "no-console": "off"
  }
};
```

**What did you do? Please include the actual source code causing the issue.**

I have a basic test case of the problem in this code:

```js
const props = {
  name: "Eric",
  age: 30,
  occupation: "Artist"
}

let { name, age } = props;
age = 3;

console.log(name);
console.log(age);
```

ESLint says that `let { name, age } = props;` is a problem and should use `const`. 

Here, two variables are created via destructuring syntax, and one is later reassigned but the other is not. 

Fixing the code as the rule suggests means using `const`, which creates a syntax error.

**What did you expect to happen?** 

I'm not sure :)

**What actually happened? Please include the actual, raw output from ESLint.**

```
$ ./node_modules/.bin/eslint app.js

/Users/eric/Desktop/eslint/app.js
  7:7  error  'name' is never reassigned. Use 'const' instead  prefer-const

✖ 1 problem (1 error, 0 warnings)
```

**Comment by ljharb** [comment]
The proper way to fix this is:
```js
const { name } = props;
let { age } = props;
```

**Comment by not-an-aardvark** [comment]
Does the [`destructuring: "all"`](http://eslint.org/docs/rules/prefer-const#destructuring) option solve your issue?

**Comment by ericandrewlewis** [comment]
Sorry, I made an over-simplified example that missed my problem.

In the case of using a rest operator, where you want to get an object that is only some properties of an object.

```
const props = {
  name: "Eric",
  age: 30,
  occupation: "Artist"
}

let { name, ...otherProps } = props;
otherProps = 3;

console.log(name);
console.log(otherProps);
```

I just tried `destructuring: "all"` with this example but I still get the error.

**Comment by vitorbal** [comment]
@ericandrewlewis apologies for the delay. Thank you for clarifying your example. I tried it locally and it does indeed report an error, even with the `destructuring: "all"` option set. I believe this is a bug. I'll work on this.

**Comment by ericandrewlewis** [comment]
🎊
